static/app/views/dashboardsV2/widgetCard.tsx

```diff
@@ -140,7 +130,7 @@ class WidgetCard extends React.Component<Props> {
                 eventName: 'Dashboards2: Table Widget - Open in Discover',
                 organization_id: parseInt(this.props.organization.id, 10),
               });
-              browserHistory.push(eventView.getResultsViewUrlTarget(organization.slug));
+              openDashboardWidgetQuerySelectorModal({organization, widget});
```

Comment:
This opens the dashboard selector modal for table/big number etc widgets which can only have one query and for chart widgets with a single query. We should have these go straight to discover right? Not sure what the design is on that, but it doesn't match the "Multiple queries were used to create this widget visualization. Which query would you like to view in Discover?" description in the modal

Reply:
Also, this should be feature gated by the `connect-discover-and-dashboards` flag for the table widget right?

Reply:
I think you're right, I've updated to just redirect the user to Discover when the widget only has 1 query. Since table widgets only have 1 query, the redirect should be immediate without the modal pop up.